lib/pure/asyncftpclient.nim

```diff
@@ -75,12 +75,57 @@
 ##      waitFor(main())
 
 
-import asyncdispatch, asyncnet, strutils, parseutils, os, times
-
-from ftpclient import FtpBaseObj, ReplyError, FtpEvent
+import asyncdispatch, asyncnet, nativesockets, strutils, parseutils, os, times
 from net import BufferSize
 
 type
+  FtpBase*[SockType] = ref FtpBaseObj[SockType]
+  FtpBaseObj*[SockType] = object
```

Comment:
I would get rid of the non-async variant and simply move the fields into the ``AsyncFtpClient``. You can also remove the ``*Obj`` variants I think.